spec/message.abnf

```diff
@@ -0,0 +1,58 @@
+message = [s] *(declaration [s]) body [s]
+
+declaration = let s variable [s] "=" [s] "{" [s] expression [s] "}"
```

Comment:
In writing a response to [this comment](https://github.com/unicode-org/message-format-wg/issues/350#issuecomment-1433903054), I noticed that we do not permit nested expressions.

My example wanted to look something like this:

```
let 1901 = :eq var1=|1901| var2={:locale_get key=variant}
```

Reply:
I'm assuming you mean something like the following, using the current syntax?

    let $year = {:eq var1=1901 var2={:locale_get key=variant}}

Or perhaps this?

    let $year = {:eq var1=1901 var2={:locale_get} key=variant}

We considered this in early 2021, when we discussed the EM and EZ data models. The EZ model proposed something we dubbed the "treeness" of function calls. I did a presentation in August 2021 attempting to break the deadlock: [The Third Data Model](https://docs.google.com/presentation/d/1pex8lEIQ0dFs72ATxva0IprIP6xLe14Sop2Oy4xPXFo/edit?usp=sharing). The group agreed to move forward with the function call model it proposed:

* function calls cannot be nested,
* variables passed into functions would have their own data model nodes, i.e. `{kind: "Variabled", name: "foo"}` rather than be just strings resolved inside functions (like the EM proposal wanted).

Reply:
One significant part of that earlier conversation was to identify that local variables may be used to get the same effect, as in:
```
let $var2 = {:locale_get key=variant}
let $year = {:eq var1=1901 var2={$var2}}
```

Reply:
thanks @stasm and @eemeli. We should call out "no nesting" in the spec. In fact, I'll task myself with that in #353.